manager/controlapi/task.go

```diff
@@ -110,7 +110,14 @@ func (s *Server) ListTasks(ctx context.Context, request *api.ListTasksRequest) (
 				if name == "" {
 					// If Task name is not assigned then calculated name is used like before.
 					// This might be removed in the future.
-					name = fmt.Sprintf("%v.%v.%v", e.ServiceAnnotations.Name, e.Slot, e.ID)
+					// We use the following scheme for Task name:
+					// Name := <TaskID>.<Slot>.<ServiceAnnotations.Name> (replicated mode)
+					//      := <TaskID>.<NodeID>.<ServiceAnnotations.Name> (global mode)
+					if e.Slot != 0 {
+						name = fmt.Sprintf("%v.%v.%v", e.ID, e.Slot, e.ServiceAnnotations.Name)
+					} else {
+						name = fmt.Sprintf("%v.%v.%v", e.ID, e.NodeID, e.ServiceAnnotations.Name)
+					}
```

Comment:
Thanks @nishanttotla. The PR has been updated. Please take a look and let me know if there are any issues.